### PR TITLE
cut/copy/paste via shortcuts has been corrected. Earlier, event wasn't

### PR DESCRIPTION
**Pull request by shapon2004**
cut/copy/paste via shortcuts has been corrected. Earlier, event wasn't fired on empty list.

### CODE REVIEW

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/FolderTreeRenderer.java [hunk 1]
```diff
@@ -127,7 +129,38 @@ public void onEvent(Event event) throws Exception {
...
+    dataRow.setDroppable("true");
+	dataRow.addEventListener(Events.ON_DROP, new EventListener<DropEvent>() {
+		@Override
+		public void onEvent(DropEvent event) throws Exception {
+
+			try {
+				FolderTreeNode dropedToItem = ((Treeitem) event.getTarget().getParent()).getValue();
+				FolderType selectedFolder = (FolderType) dropedToItem.getData();
+
+				Object dropedObject = null;
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/FolderTreeRenderer.java]: `dropedObject` should be `droppedObject`.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/FolderTreeRenderer.java [hunk 2]
```diff
@@ -127,7 +129,38 @@ public void onEvent(Event event) throws Exception {
...
+					FolderTreeNode draggedItem = ((Treeitem) event.getDragged().getParent()).getValue();
+					dropedObject = draggedItem.getData();
+				}
+
+				if (selectedFolder != null && dropedObject != null) {
+					mainC.getBaseListboxController().drop(selectedFolder, dropedObject,
+							mainC.getPortalSession().getCurrentFolder());
+				}
+			} catch (Exception e) {
+				LOGGER.error("Error Occured in Drag and Drop on Tree", e);
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/FolderTreeRenderer.java]: "Occured" should be "occurred".

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BaseListboxController.java [hunk 3]
```diff
@@ -621,6 +621,15 @@ public void paste() throws Exception {
     copyAndPasteController.paste(currentFolder);
     refreshContent();
   }
+  
+	public void drop(FolderType dropToFolder,Object droppObject, FolderType currentFolder) throws Exception {
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BaseListboxController.java]: `droppObject` should be `dropObject`.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/renderer/SummaryItemRenderer.java [hunk 4]
```diff
@@ -276,6 +284,29 @@ public void onEvent(Event event) throws Exception {
             }
 
         });
+        
+        listitem.addEventListener(Events.ON_DROP, new EventListener<DropEvent>() {
+            @Override
+            public void onEvent(DropEvent event) throws Exception {
+            	try {
+            	Listitem droppedToItem = (Listitem)event.getTarget();
+            	Object dropedObject=null;
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/renderer/SummaryItemRenderer.java]: `dropedObject` should be `droppedObject`.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/CopyAndPasteController.java [hunk 5]
```diff
@@ -344,4 +344,45 @@ public void paste(FolderType currentFolder) throws Exception {
...
+				} else if (ItemHelpers.isOwner(this.currentUser, currentFolder)) {
+					isCut = true;
+					updateSelectedItems(selections);
+				} else {
+					Notification.error(Labels.getLabel("portal_onlyOwnerCanCutFromCurrent_message"));
+				}
+			} catch (Exception e) {
+				Messagebox.show(Labels.getLabel("portal_failedCut_message"), "Apromore", Messagebox.OK,
+						Messagebox.ERROR);
+				LOGGER.error(e.getMessage());
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/CopyAndPasteController.java]: It'd be better to have a stack trace.  Perhaps `LOGGER.error("Unable to perform Cut", e);`?

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/CopyAndPasteController.java [hunk 6]
```diff
@@ -344,4 +344,45 @@ public void paste(FolderType currentFolder) throws Exception {
...
+		try {
+			if (isCut) {
+				selectedTargetFolderId = currentFolder.getId();
+				moveSelectedItems();
+				Notification.info(Labels.getLabel("portal_itemsDropped_message"));
+				clearSelectedItems();
+			}
+		} catch (Exception e) {
+			Messagebox.show(Labels.getLabel("portal_failedPaste_message"), "Apromore", Messagebox.OK, Messagebox.ERROR);
+			LOGGER.error(e.getMessage());
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/CopyAndPasteController.java]: Again, it'd be better to have a stack trace, e.g. `LOGGER.error("Unable to perform Paste", e);`

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/FolderTreeRenderer.java [hunk 7]
```diff
@@ -127,7 +129,38 @@ public void onEvent(Event event) throws Exception {
...
-  }
+    
+    dataRow.setDraggable("true");
+    dataRow.setDroppable("true");
+	dataRow.addEventListener(Events.ON_DROP, new EventListener<DropEvent>() {
+		@Override
+		public void onEvent(DropEvent event) throws Exception {
+
+			try {
+				FolderTreeNode dropedToItem = ((Treeitem) event.getTarget().getParent()).getValue();
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/FolderTreeRenderer.java]: `dropedToItem` should be `droppedToItem`.   Or perhaps `dropTarget`?

**Review by raboczi — CHANGES_REQUESTED** [review]
Minor changes requested; mainly typos and some fussiness about logging.

**Review by shapon2004 — COMMENTED** [review]
Done

**Review by shapon2004 — COMMENTED** [review]
Done

**Review by raboczi — APPROVED** [review]
All requested changes addressed; approved.

**Pull request merged**